sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -70,12 +70,28 @@ class SparkSqlAstBuilder extends AstBuilder {
 
   /**
    * Create a [[ShowTablesCommand]] logical plan.
+   * Example SQL :
+   * SHOW TABLES [(IN|FROM) database_name] ['identifier_with_wildcards']
    */
   override def visitShowTables(ctx: ShowTablesContext): LogicalPlan = withOrigin(ctx) {
-    if (ctx.LIKE != null) {
-      logWarning("SHOW TABLES LIKE option is ignored.")
+    if (ctx.qualifiedName() != null) {
+      logWarning("SHOW TABLES qualifiedName is ignored.")
```

Comment:
@hvanhovell Yeah.. its supported in their parser. But for every input i see a empty result set returned. Need to check if we break some test-case which expects an empty result set on a input qualified name. Let me run the test suites now :-)